src/targets/gpu/CMakeLists.txt

```diff
@@ -204,6 +204,12 @@ if(MIGRAPHX_ENABLE_MLIR)
     target_link_libraries(migraphx_gpu PRIVATE rocMLIR::rockCompiler)
 endif()
 
+math(EXPR hip_VERSION_FLAT "(${hip_VERSION_MAJOR} * 1000 + ${hip_VERSION_MINOR}) * 100000 + ${hip_VERSION_PATCH}")
+if(${hip_VERSION_FLAT} GREATER_EQUAL 500700000)
+    set(MIGRAPHX_HIP_REQUIRE_UNIFORM_WG ON CACHE BOOL "Whether HIP requires uniform workgroup sizes or not")
+    target_compile_definitions(migraphx_gpu PUBLIC -DMIGRAPHX_HIP_REQUIRE_UNIFORM_WG)
+endif()
```

Comment:
This should be checked at runtime, and it should check if the `-fno-offload-uniform-block` flag can be passed to hiprtc by trying to compile a sample program instead of using versions which can be problematic.

Reply:
I am actually working on adding support to check for flags as runtime to work with disabling newer warnings.

Reply:
Runtime check is better. For hip-clang i think we can just make use of `check_cxx_compiler_flag` but i wasn't sure how to do it for the hiprtc.  

For the runtime check for the hiprtc, how would that work ? because it is runtime compilation and wouldn't it need to check for each and every compile ? not sure if can be cached somehow

Reply:
See #2168, I added the function `hip_has_flags` to check for a warning, which works for both hipclang and hiprtc. You could use the same to check for non-uniform blocks as well. 

> wouldn't it need to check for each and every compile ?

You can assign it to a static variable so it only runs the check once.

Reply:
Okay. Made use of #2168